WordPress/wordcamp.org
# Blocks: Update packages, fix deprecations & new configurations

Update the `@wordpress/*` packages, which includes some configuration changes and component prop deprecations. Remove deprecated `isDefault` prop from `<Button />`. Update eslint config to use `@wordpress/eslint-plugin/recommended-with-formatting`, which uses eslint for formatting rules instead of prettier. This prevents the prettier ruleset from running before we're ready to use it. Lastly, this adds the prettier config & format command, so we can test out whether we want to adopt the auto-formatting features.

### How to test the changes in this Pull Request:

1. Run `yarn` to update local packages
2. Building, testing, etc should all work
3. Linters should pass
4. Try running `yarn workspace wordcamp-blocks run format:js` to test out the prettier commands
